### PR TITLE
Added authorization function

### LINKED ISSUE #179: Add a Github OAuth Custom Authorizer Lambda Function that the Gateway API can use
1. Takes a token and a request
2. Validates Token from Github
3. Completes the request/succeeds if the token is valid

### PR DESCRIPTION
**Pull request by dane-johnson**
Closes #179

### CODE REVIEW

File: awslambda/authorizer/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+exports.handler = (event, context, callback) => {
+  const token = event.authorizationToken;
+  const opts = httpsOpts(token);
+  axios.get('https://api.github.com/', opts)
+    .then(() => {
+      callback(null, generatePolicy('user', 'Allow', event.methodArn))
+    })
+    .catch(() => {
+      callback('Unauthorized')
+    })
```
Comment (solkaz) [inline awslambda/authorizer/index.js]: The `catch` function should be the second argument to the `then`. https://developers.google.com/web/fundamentals/getting-started/primers/promises#promise-terminology

Comment (dane-johnson) [inline awslambda/authorizer/index.js]: This is the way we've always done it with axios.
https://github.com/mzabriskie/axios#example

Comment (solkaz) [inline awslambda/authorizer/index.js]: We've migrated away from that pattern as isn't necessarily correct. https://developers.google.com/web/fundamentals/getting-started/primers/promises#chaining

Comment (solkaz) [inline awslambda/authorizer/index.js]: If you look in the SnippetArea container and the Auth component, you will see that pattern being applied.

Comment (dane-johnson) [inline awslambda/authorizer/index.js]: From the Google article.
> You can also use catch():
> 
> `get('story.json').then(function(response) {
>   console.log("Success!", response);
> }).catch(function(error) {
>   console.log("Failed!", error);
> })`
> 
> There's nothing special about catch(), it's just sugar for then(undefined, func), but it's more readable. Note that the two code examples above do not behave the same, the latter is equivalent to:
> 
> `get('story.json').then(function(response) {
>   console.log("Success!", response);
> }).then(undefined, function(error) {
>   console.log("Failed!", error);
> })`
> 
Nothing incorrect about it. Just a way of separating success and failure.

Comment (solkaz) [inline awslambda/authorizer/index.js]: What I'm saying is that the two should be bundled together, as they're concerned with the same function call. But that's trivial I guess

**Pull request merged**